pkg/buildid/buildid.go

```diff
@@ -28,12 +27,8 @@ import (
 	"github.com/parca-dev/parca-agent/pkg/elfreader"
 )
 
-func BuildID(path string) (string, error) {
-	f, err := elf.Open(path)
-	if err != nil {
-		return "", fmt.Errorf("failed to open elf: %w", err)
-	}
-
+// BuildID reads the build ID from an object file.
+func BuildID(path string, f *elf.File) (string, error) {
```

Comment:
We should also eliminate the use of the `path` here. I think can be read from `file.Name` or similar.

```suggestion
func BuildID(f *elf.File) (string, error) {
```

Reply:
I checked the source code of the elf package and couldn't find a way to get the path, that's why the solution is not elegant :(

Reply:
If we pass the file that the path points to, we still have that information in the file, no? Or does the path point to something else?